state/state_test.go

```diff
@@ -90,6 +90,32 @@ func (s *StateSuite) SetUpTest(c *gc.C) {
 	}
 }
 
+func (s *StateSuite) TestOpenController(c *gc.C) {
+	controller, err := state.OpenController(state.OpenParams{
+		Clock:              clock.WallClock,
+		ControllerTag:      s.State.ControllerTag(),
+		ControllerModelTag: s.modelTag,
+		MongoInfo:          statetesting.NewMongoInfo(),
+		MongoDialOpts:      mongotest.DialOpts(),
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(controller.Close(), gc.IsNil)
+}
+
+func (s *StateSuite) TestOpenControllerTwice(c *gc.C) {
+	for i := 0; i < 2; i++ {
+		controller, err := state.OpenController(state.OpenParams{
```

Comment:
Nit: state.OpenParams could be pulled out into a testOpenParams function or similar to avoid duplication...